packages/data-store/src/utils/digitalCredential/MappingUtils.ts

```diff
@@ -0,0 +1,112 @@
+import { AddDigitalCredentialArgs } from '../../types/digitalCredential/IAbstractDigitalCredentialStore'
+import { DigitalCredentialEntity } from '../../entities/digitalCredential/DigitalCredentialEntity'
+import {
+  CredentialMapper,
+  decodeSdJwtVc,
+  DocumentFormat,
+  IVerifiableCredential,
+  IVerifiablePresentation,
+  SdJwtDecodedVerifiableCredentialPayload,
+} from '@sphereon/ssi-types'
+import {
+  CredentialDocumentFormat,
+  CredentialType,
+  DigitalCredential,
+  NonPersistedDigitalCredential,
+} from '../../types/digitalCredential/digitalCredential'
+import { computeEntryHash } from '@veramo/utils'
+import { createHash } from 'crypto'
+
+function determineCredentialType(raw: string): CredentialType {
+  if (CredentialMapper.hasProof(raw)) {
+    if (CredentialMapper.isCredential(raw)) {
+      return CredentialType.VC
+    } else if (CredentialMapper.isPresentation(raw)) {
+      return CredentialType.VP
+    }
+  } else {
+    if (CredentialMapper.isCredential(raw)) {
+      return CredentialType.C
+    } else if (CredentialMapper.isPresentation(raw)) {
+      return CredentialType.P
+    }
+  }
+  throw new Error(`Couldn't determine the type of the credential: ${raw}`)
+}
+
+function determineCredentialDocumentFormat(documentFormat: DocumentFormat): CredentialDocumentFormat {
+  switch (documentFormat) {
+    case DocumentFormat.JSONLD:
+      return CredentialDocumentFormat.JSON_LD
+    case DocumentFormat.JWT:
+      return CredentialDocumentFormat.JWT
+    case DocumentFormat.SD_JWT_VC:
+      return CredentialDocumentFormat.SD_JWT
+    default:
+      throw new Error(`Not supported document format: ${documentFormat}`)
+  }
+}
+
+function getExpiresAt(uniformDocument: IVerifiableCredential | IVerifiablePresentation | SdJwtDecodedVerifiableCredentialPayload): Date | undefined {
+  if ('expirationDate' in uniformDocument) {
+    return new Date(uniformDocument.expirationDate)
+  } else if ('exp' in uniformDocument) {
+    return new Date(uniformDocument.exp)
+  }
+  return undefined
+}
+
+function getIssuedAt(uniformDocument: IVerifiableCredential | IVerifiablePresentation | SdJwtDecodedVerifiableCredentialPayload): Date | undefined {
+  if ('issuanceDate' in uniformDocument) {
+    return new Date(uniformDocument.issuanceDate)
+  } else if (uniformDocument.proof && CredentialMapper.getFirstProof(uniformDocument as IVerifiableCredential | IVerifiablePresentation)) {
+    return new Date(CredentialMapper.getFirstProof(uniformDocument as IVerifiableCredential | IVerifiablePresentation)!.created)
+  } else if ('validFrom' in uniformDocument) {
+    return new Date(uniformDocument['validFrom'])
+  } else if ('iat' in uniformDocument) {
+    return new Date(uniformDocument['iat'] * 1000)
+  }
+
+  return undefined
```

Comment:
Issuance date can never be undefined and thus should throw an error

Reply:
this can be undefined because we can pass unsigned documents to it.

Reply:
Signing has nothing to do with validFrom (previous Issuance)dates/times

Reply:
issuanceDate (validFrom) is optional in VCDM 2. but does your comment mean that I can't use the signature date as issuanceDate? if so I have to alter my function for getting that

Reply:
issuanceDate (validFrom) is optional in VCDM 2.

Reply:
No you shouldn't be using that. VCs can have multiple signatures. There is a reason for the name change to begin with

Reply:
yeah removed that part of the logic